core/services/vrf/listener_v2.go

```diff
@@ -672,7 +672,9 @@ func (lsn *listenerV2) checkReqsFulfilled(ctx context.Context, l logger.Logger,
 			return fulfilled, errors.Wrap(err, "creating getCommitment payload")
 		}
 
-		reqBlockNumber := new(big.Int).SetUint64(req.req.Raw.BlockNumber)
+		// Subtract 5 since the newest block likely isn't indexed yet and will cause "header not
+		// found" errors.
+		reqBlockNumber := new(big.Int).SetUint64(req.req.Raw.BlockNumber - 5)
```

Comment:
This is interesting - so the behavior we ended up seeing was that the request block number is > the latest head number?

If this is the case, then I think we should add an additional check to `getAndRemoveConfirmedLogsBySub`:

```golang
// current:
if r := lsn.reqs[i]; r.confirmedAtBlock <= latestHead {
// to:
if r := lsn.reqs[i]; r.confirmedAtBlock <= latestHead && r.req.Raw.BlockNumber <= latestHead {
```

Though, thinking about it a little bit more, I'm not sure how it's possible we even get this request included to process if we `r.confirmedAtBlock <= latestHead` and we have `r.confirmedAtBlock > r.req.Raw.BlockNumber` (the confirmed at block must be greater than the request's block number, since we have a minimum of 1 confirmation across all chains, though most chains have 3), so this additional check _should_ be redundant.

Reply:
Oh good catch, I meant to subtract 5 from curBlock not reqBlockNumber.

I think reqBlockNumber > latest head is what this code is originally meant to fix. In that case it doesn't matter if we get an error since we know request is unfulfilled.